### PR TITLE
[@types/codemirror] Fix types

### PR DESCRIPTION
**Pull request by orblazer**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://codemirror.net/doc/manual.html#config>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Hello,
I have fix the types for correspond to the documentation.
Some addons of codemirror is remaining missing type.

Thanks.

### CODE REVIEW

**Comment by mihailik** [comment]
Wow, this is a large PR now!

Thanks for lots of useful updates, @orblazer. **Quite a lot** of good reasonable fixes and improvements here. Awesome work!!

I've had a quick glance, there were a few things I may not quite agree with there. In addition to my previous comments, please consider these broader points:

* it's best NOT to rename anything, unless a very good reason exists
* not sure if `HTMLElement` should be replaced with `Element`

I will review the changes in detail, but overall quite impressed. Great care on docs and re-formatting too. Thanks a lot again!

**Comment by mihailik** [comment]
On the point of renaming: it's OK to rename obvious typo/spelling error.

For things like `Lint` --> `LintHelper`, it may be much helpful to stick with existing name. The reason here is that people would already rely old names, and use in **their code**.

When they update `CodeMirror` typings, we should minimise any churn they experience as result. Especially if they support somebody else's code, or libraries built on top of `CodeMirror` -- updating that legacy/external code with `LintHelper` can be very very painful. Let's try to avoid it as much as possible.

Given the amount of good care and improvement you've put into these changes, it's even more important we make it smooth for people to adopt newer typings.

**Review by mihailik — CHANGES_REQUESTED** [review]

File: types/codemirror/addon/comment/continuecomment.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Type definitions for CodeMirror
+// Project: https://github.com/codemirror/CodeMirror
+// Definitions by: orblazer <https://github.com/orblazer>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// See docs https://codemirror.net/doc/manual.html#active-line
```
[the remark this answers is not in the document]
Comment (orblazer) [reply types/codemirror/addon/comment/continuecomment.d.ts]: Oh, wes i have just copy-paste and missing replace url, i go fix that in few minutes

**Comment by orblazer** [comment]
Hello @mihailik, that PR is now OK ?

**Comment by orblazer** [comment]
The pr is now here : https://github.com/DefinitelyTyped/DefinitelyTyped/pull/35456

**Comment by orblazer** [comment]
The pr is now here : https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38033

**Pull request closed without merging**